cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/AbstractChunkedLogProducer.java

```diff
@@ -65,6 +65,7 @@ public ByteBuf nextChunk() throws Exception {
     if (!hasStarted) {
       hasStarted = true;
       startBuffer = Unpooled.copiedBuffer(onWriteStart());
+      return startBuffer;
```

Comment:
can you add comment on how this fixes the issue ? also there is no comment on this class, makes it hard to understand whats happening, I understand that you didn't write this class, but if you can add some javadoc, it will be useful to understand the behavior in future.

Reply:
It is a implementation of `BodyProducer#nextChunk()`. It already has a javadoc